astrokat/noisediode.py

```diff
@@ -68,13 +73,21 @@ def on(kat,
     user_logger.trace('TRACE: nd on at {} ({})'
                       .format(timestamp,
                               time.ctime(timestamp)))
-    msg = ('Switch noise-diode on at {}'
+    msg = ('Request: Switch noise-diode on at {}'
            .format(timestamp))
     user_logger.info(msg)
 
     replies = kat.ants.req.dig_noise_source(timestamp, 1)
     if not kat.dry_run:
-        _katcp_reply_to_log_(replies)
+        timestamp = _katcp_reply_to_log_(replies)
+    else:
+        # - use integer second boundary as that is most likely be an exact
+        #   time that DMC can execute at
+        msg = ('Dry-run: Set all noise diodes with timestamp {} ({})'
+               .format(np.ceil(timestamp),
```

Comment:
If the caller passed in a timestamp, if might have a fractional part, and we assume they know what they are doing.  We should log that time.  The DMC will try to get close to that time, even if it isn't on an integer boundary.

(Also note that the ctime calling isn't using the `ceil` of the timestamp, so it could differ slightly - we should be consistent)